codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -199,6 +195,21 @@ private static void encode3to4(
         }
     }
 
+    // package-private for testing
+    static int encodedBufferSize(int len, boolean breakLines) {
+        // Cast len to long to prevent overflow
+        long len43 = ((long) len << 2) / 3;
+
+        // Account for padding
+        long ret = (len43 + 3) & ~3;
+
+        if (breakLines) {
+            ret += len43 / MAX_LINE_LENGTH;
+        }
+
+        return ret < Integer.MAX_VALUE ? (int) ret : Integer.MAX_VALUE;
```

Comment:
if there is a chance it may succeed then that sgtm ... if we know it will fail and are relying on `IndexOutOfBoundsException` to be thrown later we should just throw not instead of wasting resources.